build broken when sphinx spelling extension not installed

The build is currently broken on systems with sphinx but no `sphinxcontrib.spelling` extension installed. `configure` only looks for `sphinx`, so docs are enabled, but the build fails here:
```
/usr/tce/bin/python3 -m sphinx -M man . ./ 
Running Sphinx v1.8.4

Extension error:
Could not import extension sphinxcontrib.spelling (exception: No module named 'sphinxcontrib.spelling')
make[2]: *** [man1/flux.1] Error 2
make[2]: Leaving directory `/g/g0/grondo/git/flux-core.git/doc'
make[1]: *** [all-recursive] Error 1
make[1]: Leaving directory `/g/g0/grondo/git/flux-core.git/doc'
make: *** [all-recursive] Error 1
```

--- a/doc/conf.py
+++ b/doc/conf.py
@@ -39,18 +39,6 @@ release = '0.13.0'
 
 # -- General configuration ---------------------------------------------------
 
-# Add any Sphinx extension module names here, as strings. They can be
-# extensions coming with Sphinx (named 'sphinx.ext.*') or your custom
-# ones.
-extensions = [
-    'sphinxcontrib.spelling',
-]
-
-# sphinxcontrib.spelling settings
-spelling_word_list_filename = [
-    'test/spell.en.pws'
-]
-
 # Add any paths that contain templates here, relative to this directory.
 templates_path = ['_templates']
 

--- a/doc/requirements.txt
+++ b/doc/requirements.txt
@@ -1,3 +1,2 @@
 sphinx
 sphinx-rtd-theme
-sphinxcontrib-spelling

